src/loaddb/load_db.c

```diff
@@ -965,8 +965,8 @@ ldr_exec_query_from_file (const char *file_name, FILE * input_stream, int *start
       stmt_cnt = db_parse_one_statement (session);
       if (stmt_cnt > 0)
 	{
-	  db_get_parser_line_col (session, &parser_end_line_no, NULL);
 	  stmt_id = db_compile_statement (session);
+	  parser_end_line_no = db_get_line_of_last_statement (session, stmt_id);
```

Comment:
when the latest statement has an error, its stmt_id is < 0 and parser_end_line_no is reset to 0.
this means parser_end_line_no is set only for a valid statement.
Do you want this?